nixos/lib/testing/nodes.nix

```diff
@@ -3,27 +3,34 @@ testModuleArgs@{ config, lib, hostPkgs, nodes, ... }:
 let
   inherit (lib) mkOption mkForce optional types mapAttrs mkDefault mdDoc;
 
+  inherit (import ../.. {}) evalSystemConfiguration baseModules;
+
   system = hostPkgs.stdenv.hostPlatform.system;
 
+  testModules = [
+    ./nixos-test-base.nix
+    { key = "nodes"; _module.args.nodes = config.nodesCompat; }
+    ({ config, ... }:
+      {
+        virtualisation.qemu.package = testModuleArgs.config.qemu.package;
+
+        # Ensure we do not use aliases. Ideally this is only set
+        # when the test framework is used by Nixpkgs NixOS tests.
+        nixpkgs.config.allowAliases = false;
+      })
+    testModuleArgs.config.extraBaseModules
+  ];
+
   baseOS =
-    import ../eval-config.nix {
-      inherit system;
+    evalSystemConfiguration {
       inherit (config.node) specialArgs;
-      modules = [ config.defaults ];
-      baseModules = (import ../../modules/module-list.nix) ++
-        [
-          ./nixos-test-base.nix
-          { key = "nodes"; _module.args.nodes = config.nodesCompat; }
-          ({ config, ... }:
-            {
-              virtualisation.qemu.package = testModuleArgs.config.qemu.package;
-
-              # Ensure we do not use aliases. Ideally this is only set
-              # when the test framework is used by Nixpkgs NixOS tests.
-              nixpkgs.config.allowAliases = false;
-            })
-          testModuleArgs.config.extraBaseModules
-        ];
+      modules = [
+        { nixpkgs = { inherit system; }; }
+        config.defaults
+        testModules
+        # TODO (during review): clarify what contract is being actually consumed here (and replace this line with an explanation)
+        { _module.args.baseModules = lib.mkForce (baseModules ++ testModules); }
```

Comment:
Ok, I guess I actually found it later myself: `nixosOptionsDoc`.

If that'd the only contract consumer, this then seems misplaced and superfluous inside `nixos/lib/testing/nodes.nix`. Or am I missing something?